layer/unittest/trace_event_log_tests.cc

```diff
@@ -113,9 +276,180 @@ TEST(TraceEvent, CompleteEventCreation) {
       std::find_if(trace_args.begin(), trace_args.end(),
                    [](Attribute *arg) { return arg->isa<DurationAttr>(); });
   ASSERT_TRUE(dur_it != trace_args.end());
-  EXPECT_EQ((*dur_it)->cast<DurationAttr>()->GetValue().ToNanoseconds(),
+  ASSERT_EQ((*dur_it)->cast<DurationAttr>()->GetValue().ToNanoseconds(),
             duration.ToNanoseconds());
 }
 
+TEST(TraceEvent, InstantEventToString) {
+  InstantEvent instant_event("compile_time_init", "compile_time", 123, 321);
+  std::ostringstream expected_str;
+  expected_str << "{ " << std::quoted("name") << " : "
+               << std::quoted("compile_time_init") << ", " << std::quoted("ph")
+               << " : " << std::quoted("i") << ", " << std::quoted("cat")
+               << " : " << std::quoted("compile_time") << ", "
+               << std::quoted("pid") << " : " << 123 << ", "
+               << std::quoted("tid") << " : " << 321 << ", "
+               << std::quoted("ts") << " : " << std::fixed
+               << ToUnixMillis(instant_event.GetCreationTime().GetValue())
+               << ", " << std::quoted("s") << " : " << std::quoted("g") << ", "
+               << std::quoted("args") << " : "
+               << "{ " << std::quoted("scope") << " : " << std::quoted("g")
+               << " } },";
```

Comment:
I find this difficult to parse, there are many moving parts and calls here. Could we convert this to a [raw string literal](https://learn.microsoft.com/en-us/cpp/cpp/string-and-character-literals-cpp?view=msvc-170#raw-string-literals-c11)? E.g.: `string_view expected_str = R"({ "name": ...},)";`

if that's too long, you can add it in several chunks with `<<`, but it would be better to keep it as close to the actual string as possible.

Because creation time is not fixed, we can set it to some fixed value before printing.